packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/AmplifyPushNotificationBroadcastReceiver.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.amplify.amplify_push_notifications
+
+
+import android.annotation.SuppressLint
+import android.app.ActivityManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.amplifyframework.pushnotifications.pinpoint.utils.PushNotificationsUtils
+import com.google.firebase.messaging.RemoteMessage
+
+// TODO: This BroadcastReceiver needs to be replaced by the Firebase Service
+class PushNotificationReceiver : BroadcastReceiver() {
+    companion object {
+        private val TAG = "PushNotificationReceiver"
+    }
+
+    @SuppressLint("LongLogTag")
+    override fun onReceive(context: Context, intent: Intent?) {
+        intent?.let {
+            if (isPushNotificationIntent(it)) {
+                val remoteMessage = RemoteMessage(it.extras)
+                val remoteMessageBundle = getBundleFromRemoteMessage(remoteMessage)
+                if (isAppInForeground(context)) {
+                    val notificationHashMap = convertBundleToHashMap(remoteMessageBundle)
+                    Log.d(TAG, "Send foreground message received event: $notificationHashMap")
+
+                    PushNotificationEventsStreamHandler.sendEvent(
+                        PushNotificationsEvent(
+                            NativeEvent.FOREGROUND_MESSAGE_RECEIVED,
+                            notificationHashMap
+                        )
+                    )
+                } else {
+                    Log.d(TAG, "App is in background, start background service and enqueue work")
+                    try {
+
+                        val payload = getPayloadFromRemoteMessage(remoteMessage)
+                        // TODO: Check how to add a flag to indicate app was opened by a notificaiton
+                        PushNotificationsUtils(context).showNotification(
+                            payload,
+                            AmplifyPushNotificationsPlugin::class.java
+                        )
+
+                        // TODO: Start a background headless service
+//                        FlutterMain.startInitialization(context)
+//                        FlutterMain.ensureInitializationComplete(context, null)
+//                        PushNotificationBackgroundService.enqueueWork(context, it)
+
+                    } catch (exception: Exception) {
+                        Log.e(TAG, "Something went wrong while starting headless task $exception")
```

Comment:
Decide on a better error handling strategy... how do Flutter users react to these errors? Or is it okay to swallow and log?